src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/RealExternalServiceDAO.scala

```diff
@@ -0,0 +1,43 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+import org.broadinstitute.dsde.rawls.model.AgoraEntity
+import org.broadinstitute.dsde.rawls.model.JobStatus
+import org.broadinstitute.dsde.rawls.model.WorkspaceJsonSupport._
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.Try
+import spray.client.pipelining._
+import spray.http._
+import spray.http.HttpHeaders.Cookie
+import spray.httpx.SprayJsonSupport._
+import spray.json._
+
+/**
+ * @author tsharpe
+ */
+class RealExternalServiceDAO( methodRepoConf: Config, execSvcConf: Config)( implicit system: ActorSystem ) extends ExternalServiceDAO {
+
+  override def getMethodConfig( namespace: String, name: String, version: String, authCookie: HttpCookie ): Try[HttpResponse] = {
+    val url = methodRepoConf.getString("getConfigPath").format(namespace,name,version)
+    import system.dispatcher
+    val pipeline = addHeader(Cookie(authCookie)) ~> sendReceive
+    Try { Await.result(pipeline(Get(url)),Duration.Inf) }
+  }
+
+  override def getMethod( namespace: String, name: String, version: String, authCookie: HttpCookie ): Try[AgoraEntity] = {
+    val url = methodRepoConf.getString("getMethodPath").format(namespace,name,version)
+    import system.dispatcher
+    val pipeline = addHeader(Cookie(authCookie)) ~> sendReceive ~> unmarshal[AgoraEntity]
+    Try { Await.result(pipeline(Get(url)),Duration.Inf) }
+  }
+
+  override def submitJob( wdl: String, inputs: String, authCookie: HttpCookie ): Try[JobStatus] = {
```

Comment:
Interactions with the method repo should be in a different class from interactions with cromwell. If there is common code that should be in a base trait.